nixos/modules/services/cluster/druid/default.nix

```
@@ -0,0 +1,225 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.services.druid;
+  inherit (lib) concatStrings concatStringsSep mapAttrsToList concatMap attrByPath
+    mkIf mkMerge mkEnableOption mkOption concatMapStringsSep types mkPackageOptionMD;
+  druidServiceOption = serviceName: {
+    enable = mkEnableOption (serviceName);
+    restartIfChanged = mkOption {
+      type = types.bool;
+      description = ''
+        Automatically restart the service on config change.
+        This can be set to false to defer restarts on clusters running critical applications.
+        Please consider the security implications of inadvertently running an older version,
+        and the possibility of unexpected behavior caused by inconsistent versions across a cluster when disabling this option.
+      '';
+      default = false;
+    };
+    config = mkOption {
+      default = { };
+      type = types.attrsOf types.anything;
+      description = ''
+        (key=value) Configuration to be written to runtime.properties of the druid ${serviceName}
+        <https://druid.apache.org/docs/latest/configuration/index.html>
+      '';
+      example = {
+        "druid.plainTextPort" = "8082";
+        "druid.service" = "servicename";
+      };
+    };
+    jdk = mkPackageOptionMD pkgs "JDK" {
+      default = [ "jdk17_headless" ];
+    };
+    jvmArgs = mkOption {
+      type = types.str;
+      default = "";
+      description = "Arguments to pass to the JVM";
+    };
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = "Open firewall ports for ${serviceName}.";
+    };
+    internalConfig = mkOption {
+      default = { };
+      type = types.attrsOf types.anything;
+      internal = true;
+      description = "Internal Option to add to runtime.properties for ${serviceName}.";
+    };
+  };
+
+
+
+  druidServiceConfig =
+    { name
+    , serviceOptions ? cfg."${name}"
+    , allowedTCPPorts ? [ ]
+    , tmpDirs ? [ ]
+    , extraConfig ? { }
+    }: (
+      mkIf serviceOptions.enable (mkMerge [
+        {
+          systemd = {
+            services."druid-${name}" = {
+              after = [ "network.target" ];
+              description = "Druid ${name}";
+              environment = {
+                DRUID_JAVA_HOME = "${serviceOptions.jdk}";
+              };
+              wantedBy = [ "multi-user.target" ];
+              inherit (serviceOptions) restartIfChanged;
+              path = [ "${serviceOptions.jdk}" ];
```

```suggestion
              path = [ cfg.package ];
```

Path should be a list of derivations. Also, druid seems to be picking up java from the env variable, so passing it here is unnecessary. Alternatively, you can set path to `[ serviceOptions.jdk ]` and remove the DRUID_JAVA_HOME env variable.

---

@vsharathchandra could you address this comment?